src/core/variables/NumberVariable.ts

```diff
@@ -52,8 +52,9 @@ export class NumberVariable extends Variable implements INumberVariableParams {
     possibleValues?: number[],
     callback?: IVariableCallback,
   ) {
-    super(key, VariableType.NUMBER, defaultValue, callback);
+    super(key, DataType.NUMBER, defaultValue, callback);
     this.possibleValues = possibleValues ? possibleValues : [];
+    this.constraintType = ConstraintType.LIST;
```

Comment:
Good catch. I modified this to a readonly property of Variable and each class now overrides as needed.